source/iterator/snapshot_iterator.go

```diff
@@ -0,0 +1,401 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package iterator
+
+import (
+	"context"
+	"encoding/csv"
+	"errors"
+	"fmt"
+	"io"
+	"reflect"
+	"strings"
+	"time"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/jpillora/backoff"
+	marketoclient "github.com/rustiever/conduit-connector-marketo/marketo-client"
+	"github.com/rustiever/conduit-connector-marketo/source/position"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	// is the maximum number of days between two snapshots. If the gap between two snapshots is greater than this number,
+	// API will return an error. This is limitation of the API.
+	MaximumDaysGap = 744 // 31 days in Hours
+)
+
+// to iterate through the snapshots for specified configs
+type SnapshotIterator struct {
+	endpoint      string
+	fields        []string
+	client        *marketoclient.Client
+	exportID      string
+	initialDate   time.Time
+	iteratorCount int
+	errChan       chan error
+	csvReader     chan *csv.Reader
+	buffer        chan []string
+	lastMaxModied time.Time
```

Comment:
Let's please describe what these fields are. Also, typo in `lastMaxModied`.

Reply:
updated